pkg/devfile/parser/data/v2/commands_test.go

```diff
@@ -300,3 +301,84 @@ func TestDevfile200_UpdateCommands(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteCommands(t *testing.T) {
+
+	d := &DevfileV2{
+		v1.Devfile{
+			DevWorkspaceTemplateSpec: v1.DevWorkspaceTemplateSpec{
+				DevWorkspaceTemplateSpecContent: v1.DevWorkspaceTemplateSpecContent{
+					Commands: []v1.Command{
+						{
+							Id: "command1",
+							CommandUnion: v1.CommandUnion{
+								Exec: &v1.ExecCommand{},
+							},
+						},
+						{
+							Id: "command2",
+							CommandUnion: v1.CommandUnion{
+								Exec: &v1.ExecCommand{},
+							},
+						},
+						{
+							Id: "command3",
+							CommandUnion: v1.CommandUnion{
+								Composite: &v1.CompositeCommand{
+									Commands: []string{"command1", "command2", "command1"},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	tests := []struct {
+		name            string
+		commandToDelete string
+		wantCommands    []v1.Command
+		wantErr         bool
+	}{
+		{
+			name:            "Successfully delete command",
+			commandToDelete: "command1",
+			wantCommands: []v1.Command{
+				{
+					Id: "command2",
+					CommandUnion: v1.CommandUnion{
+						Exec: &v1.ExecCommand{},
+					},
+				},
+				{
+					Id: "command3",
+					CommandUnion: v1.CommandUnion{
+						Composite: &v1.CompositeCommand{
+							Commands: []string{"command1", "command2", "command1"},
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:            "Missing Command",
+			commandToDelete: "command34",
+			wantErr:         true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := d.DeleteCommand(tt.commandToDelete)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("DeleteCommand() error = %v, wantErr %v", err, tt.wantErr)
+			}
+
+			if err == nil && !tt.wantErr {
+				assert.Equal(t, tt.wantCommands, d.Commands, "The two values should be the same.")
```

Comment:
this if statement, already checked by the previous one (previous if block should return)
can directly check tt.wantCommands &  d.Commands
same for the other tests

Reply:
no we cant, because there are tests where we want an err and that should be skipped. The prev if block is to check if there is an err mis match. I've updated it to a better if else block

Reply:
it won't be skipped, because it's returning from that single test `func(t *testing.T){}`, it will continue the for loop for the rest of tests